monai/data/nifti_saver.py

```diff
@@ -95,6 +91,9 @@ def save(self, data: Union[torch.Tensor, np.ndarray], meta_data=None):
             data = data.detach().cpu().numpy()
         filename = create_file_basename(self.output_postfix, filename, self.output_dir)
         filename = f"{filename}{self.output_ext}"
+        # change data shape to be (channel, h, w, d)
+        while len(data.shape) < 4:
+            data = np.expand_dims(data, -1)
```

Comment:
Why you change a 2D data into 3D and save?

Reply:
because the `write_nifti ` can't easily distinguish (h, w, chns) vs (h, w, d). i also thought about adding a flag to `write_nifti` but it makes the interface a bit complicated.

Reply:
I think the `affine` matrix for (h, w, chns) and (h, w, d) is different, can we use it to distinguish? Or I feel it's a little bit confusing to save data in an unexpected shape. May need a discussion about it.

Reply:
> I think the `affine` matrix for (h, w, chns) and (h, w, d) is different, can we use it to distinguish? Or I feel it's a little bit confusing to save data in an unexpected shape. May need a discussion about it.

they are  the same for nifti files

Reply:
thanks for the discussion @Nic-Ma, just for clarification, I think the issue comes from, for example, writing single-channel 3D segmentation result, do we want to write array shape as
option 1 - (H, W, D, 1) or
option 2 - (H, W, D) 
both are valid (works fine with Nibabel).

Since it's segmentation prediction, one idea would be: try to match the shape of the user's input ground truth segmentation mask. But within this component, we couldn't know whether the user read the ground truth with "LoadNifti"or "LoadNifti+AddChannel" or other transforms.... in this case we should go for "always channel" (option 1) as we discussed. in this case, 2D single-channel will be (H, W, 1, 1) there's no ambiguity, but it may not match the user's ground truth. I update the docstring to reflect this.

Reply:
Hi @wyli ,

I think it's slightly different from our discussion. We discussed to suppose the input data must have `channel` dim, no matter 2D or 3D, then 2D data shape is [H, W, C], 3D data shape is [H, W, D, C] and no need to change 2D data shape to [H, W, 1, C] anymore, right?
Thanks.

Reply:
the problem is that if we save multi-channel 2D data as [H, W, C], nibabel (and also many visualisation software) will interpret it as 3D single-channel image. we should expand the dims to set the 3rd spatial dim to 1 (which is done in the NiftiSaver in this PR).

Reply:
OK, I see the problem. It makes sense to do it now.
Thanks.